src/usecase/tests/category_ucase_test.go

```diff
@@ -66,161 +67,156 @@ func TestAddCategory(t *testing.T) {
-//	cateringResult, _ := cateringRepo.GetByKey("name", "Twiist")
-//	jwt, _, _ := middleware.Passport().TokenGenerator(&middleware.UserID{userResult.ID.String()})
-//	cateringID := cateringResult.ID.String()
-//
-//	// Trying to update non-existing dish category
-//	// Should throw an error
-//	fakeID := uuid.NewV4()
-//	r.PUT("/caterings/"+cateringID+"/categories/"+fakeID.String()).
-//		SetCookie(gofight.H{
-//			"jwt": jwt,
-//		}).
-//		SetJSON(gofight.D{
-//			"name": "zxcvb",
-//		}).
-//		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
-//			assert.Equal(t, http.StatusNotFound, r.Code)
-//		})
-//
-//	// Posting new dish category to update it
-//	// Should be success
-//	r.POST("/caterings/"+cateringID+"/categories").
-//		SetCookie(gofight.H{
-//			"jwt": jwt,
-//		}).
-//		SetJSON(gofight.D{
-//			"name": "qwerty",
-//		}).
-//		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
-//			assert.Equal(t, http.StatusOK, r.Code)
-//		})
-//
-//	createdCategory, _ := categoryRepo.GetByKey("name", "qwerty", cateringID)
-//	categoryID := createdCategory.ID.String()
-//
-//	// Trying to update new dish category
-//	// Should be success
-//	r.PUT("/caterings/"+cateringID+"/categories/"+categoryID).
-//		SetCookie(gofight.H{
-//			"jwt": jwt,
-//		}).
-//		SetJSON(gofight.D{
-//			"name": "zxcvb",
-//		}).
-//		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
-//			assert.Equal(t, http.StatusNoContent, r.Code)
-//		})
+func TestDeleteCategory(t *testing.T) {
+	r := gofight.New()
+
+	categoryRepo := repository.NewCategoryRepo()
+	clientRepo := repository.NewClientRepo()
+	userRepo := repository.NewUserRepo()
+	cateringRepo := repository.NewCateringRepo()
+	clientResult, _ := clientRepo.GetByKey("name", "Dymi")
+	userResult, _ := userRepo.GetByKey("email", "admin@meals.com")
+	cateringResult, _ := cateringRepo.GetByKey("name", "Twiist")
+	jwt, _, _ := middleware.Passport().TokenGenerator(&middleware.UserID{userResult.ID.String()})
+	cateringID := cateringResult.ID.String()
+	clientID := clientResult.ID.String()
+
+	// Creates new dish category
+	// Should be success
+	r.POST("/caterings/"+cateringID+"/clients/"+clientID+"/categories").
+		SetCookie(gofight.H{
+			"jwt": jwt,
+		}).
+		SetJSON(gofight.D{
+			"name": "фингерфуд",
+		}).
+		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+			assert.Equal(t, http.StatusOK, r.Code)
+		})
+
+	createdCategory, _ := categoryRepo.GetByKey("name", "закуски", cateringID)
+	categoryID := createdCategory.ID.String()
+
+	// Trying to delete new category dish
+	// Should be success
+	r.DELETE("/caterings/"+cateringID+"/clients/"+clientID+"/categories/"+categoryID).
+		SetCookie(gofight.H{
+			"jwt": jwt,
+		}).
+		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+			assert.Equal(t, http.StatusNoContent, r.Code)
+		})
```

Comment:
Could add more test cases like adding a category for nonexisting client/catering